programs/manifest/src/program/processor/batch_update.rs

```diff
@@ -132,11 +158,67 @@ pub enum MarketDataTreeNodeType {
 }
 
 pub(crate) fn process_batch_update(
-    _program_id: &Pubkey,
+    program_id: &Pubkey,
     accounts: &[AccountInfo],
     data: &[u8],
 ) -> ProgramResult {
+    let params = BatchUpdateParams::try_from_slice(data)?;
+    process_batch_update_core(program_id, accounts, params)
```

Comment:
To make verification easier, we created entry points that take arguments directly rather than unpack them. These are the `_core` functions. We have propagated all the arguments uniformly from the existing entry point to the `_core` one. The goal was to be very clear about what the new entry point is about, so we did not want to over optimize.